galite-core/src/main/kotlin/org/kopi/galite/util/AWTToPS.kt

```diff
@@ -15,8 +15,1261 @@
+
+import kotlin.experimental.or
+
+import org.kopi.galite.util.base.InconsistencyException
+import org.kopi.galite.util.base.NotImplementedException
+
+/**
+ * A class to paint in a postscript file instead of screen
+ */
+class AWTToPS(private val stream: PrintStream, clone: Boolean) : Graphics() {
+  constructor(stream: OutputStream, clone: Boolean) : this(PrintStream(stream), clone)
+
+  constructor(stream: OutputStream) : this(stream, false) {}
+
+  fun setBoundingBox(x: Int, y: Int, width: Int, height: Int) {
+    stream.println("%%BoundingBox: $x $y $width $height")
+  }
+
+  fun setScale(x: Double, y: Double) {
+    emitScale(x, y)
+  }
+
+  /**
+   * Creates a new AWTToPS Object that is a copy of the original AWTToPS Object.
+   */
+  override fun create(): Graphics {
+    val graphics = AWTToPS(stream, true)
+    graphics.font = font
+    graphics.clippingRect = clippingRect
+    graphics.clr = clr
+    graphics.transColor = transColor
+    return graphics
+  }
+
+  /**
+   * Creates a new Graphics Object with the specified parameters,
+   * based on the original
+   * Graphics Object.
+   * This method translates the specified parameters, x and y, to
+   * the proper origin coordinates and then clips the Graphics Object to the
+   * area.
+   * @param x the x coordinate
+   * @param y the y coordinate
+   * @param width the width of the area
+   * @param height the height of the area
+   * @see translate
+   */
+  override fun create(x: Int, y: Int, width: Int, height: Int): Graphics {
+    val graphics = create()
+    graphics.translate(x, y)
+    graphics.clipRect(0, 0, width, height)
+    return graphics
+  }
+
+  /**
+   * Translates the specified parameters into the origin of
+   * the graphics context. All subsequent
+   * operations on this graphics context will be relative to this origin.
+   * @param x the x coordinate
+   * @param y the y coordinate
+   * @see scale
+   */
+  override fun translate(x: Int, y: Int) {
+    emitTranslate(x, -y)
+  }
+
+  /**
+   * Scales the graphics context. All subsequent operations on this
+   * graphics context will be affected.
+   * @param sx the scaled x coordinate
+   * @param sy the scaled y coordinate
+   * @see translate
+   */
+  fun scale(sx: Float, sy: Float) {
+    emitScale(sx.toDouble(), sy.toDouble())
+  }
+
+  /**
+   * Gets the current color.
+   * @see setColor
+   */
+  override fun getColor(): Color {
+    return clr
+  }
+
+  /**
+   * Gets the current color.
+   * @see setColor
+   */
+  fun setBackground(c: Color) {
+    backClr = c
+  }
+
+  fun setTransparentColor(trans: Color) {
+    transColor = (trans.red shl 16) + (trans.green shl 8) + trans.blue
+  }
+
+  /**
+   * Sets the current color to the specified color. All subsequent graphics operations
+   * will use this specified color.
+   * @param c the color to be set
+   * @see Color
+   *
+   * @see getColor
+   */
+  override fun setColor(c: Color?) {
+    if (c != null) {
```

Comment:
c?.let{

Reply:
why using `let`? here we dont want to return something

Reply:
using "let" doesn't require to return something ( lambda's return can be void ) and in this case you are free to keep the if ( != null ) too